azurelinuxagent/common/protocol/wire.py

```
@@ -913,49 +913,51 @@ def _call_hostplugin_with_container_check(self, host_func):
+            # Different direct channel functions report failure in different ways: by returning None, False,
+            # or raising ResourceGone or InvalidContainer exceptions.
+            if not ret:
                 logger.periodic_info(logger.EVERY_HOUR, "[PERIODIC] Request failed using the direct channel, "
-                                                        "switching to host plugin. Error: {0}".format(ustr(e)))
+                                                        "trying host plugin next.")
+        except (ResourceGoneError, InvalidContainerError) as e:  # pylint: disable=C0103
+            logger.periodic_info(logger.EVERY_HOUR, "[PERIODIC] Request failed using the direct channel, "
+                                                    "trying host plugin next. Error: {0}".format(ustr(e)))
 
-            if ret:
-                return ret
-        else:
-            logger.periodic_info(logger.EVERY_HALF_DAY, "[PERIODIC] Using host plugin as default channel.")
+        if ret and HostPluginProtocol.is_default_channel():
+            # Set direct channel as default if it succeeded.
+            logger.info("Switching default channel to WireServer.")
+            HostPluginProtocol.set_default_channel(False)
 
-        ret = self._call_hostplugin_with_container_check(host_func)
+        return ret
 
-        if not HostPluginProtocol.is_default_channel():
-            logger.info("Setting host plugin as default channel from now on. "
-                        "Restart the agent to reset the default channel.")
-            HostPluginProtocol.set_default_channel(True)
+    def send_request_using_appropriate_channel(self, direct_func, host_func):
+        primary_ret = None
+        try:
+            primary_ret = self.__send_request_using_host_channel(host_func) if HostPluginProtocol.is_default_channel() \
+                else self.__send_request_using_direct_channel(direct_func)
+        except Exception:
+            pass
```

if you swallow the exception we will lose all the debugging info that could help to determine what kind of error occurred

---

+1